packages/synapse-react-client/src/components/FilePreview/PdfPreview.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useRef } from 'react'
+import { Alert } from '@mui/material'
+import {
+  FileHandle,
+  FileHandleAssociation,
+} from '@sage-bionetworks/synapse-types'
+import { calculateFriendlyFileSize } from '../../utils/functions/calculateFriendlyFileSize'
+import { BackendDestinationEnum, getEndpoint } from '../../utils/functions'
+
+export type PdfPreviewProps = {
+  fileHandle: FileHandle
+  fileHandleAssociation: FileHandleAssociation
+}
+
+export const maxPdfSize = Math.pow(1024, 2) * 30 // 30MB
+const friendlyMaxPdfSize = calculateFriendlyFileSize(maxPdfSize) // 30MB
+
+export const getFhaUrl = (fha: FileHandleAssociation) => {
+  return `Portal/filehandleassociation?associatedObjectId=${fha.associateObjectId}&associatedObjectType=${fha.associateObjectType}&fileHandleId=${fha.fileHandleId}`
+}
```

Comment:
may want to refactor to share logic with `SynapseClient.getPortalFileHandleServletUrl`

Reply:
Updated!